nexusutils/dataconverter/readers/rii_database/reader.py

```diff
@@ -37,77 +43,307 @@ def read_yml_file(filename: str) -> Dict[Any, Any]:
     return yml_file
 
 
-def read_dispersion(filename: str, identifier: str = 'dispersion_x') -> Dict[str, Any]:
-    """Reads rii dispersions from yaml files"""
+def is_table_k_formula_n_disp(dispersion1: dict, dispersion2: dict):
+    """Checks if the two dispersions are one tabular k formula and one formula n"""
+    if "tabulated k" == dispersion1["type"] and dispersion2["type"].startswith(
+        "formula"
+    ):
+        return True
+
+    if "tabulated k" == dispersion2["type"] and dispersion1["type"].startswith(
+        "formula"
+    ):
+        return True
+
+    return False
+
+
+def read_metadata(yml_file: dict) -> Dict[str, Any]:
+    """Reads metadata from a dispersion yaml file"""
     entries: Dict[str, Any] = {}
+    if "REFERENCES" in yml_file:
+        pass
+
+    if "COMMENTS" in yml_file:
+        pass
```

Comment:
Are these if passes necessary? Or is this to be filled up later?

Reply:
This I'll add later in a new PR. Right now I don't parse any metadata, yet.